akka-projection-core/src/test/scala/akka/projection/internal/metrics/InternalProjectionStateMetricsSpec.scala

```diff
@@ -0,0 +1,382 @@
+/*
```

Comment:
This file contains the supporting code that's necessary to write all the `XyzMetricSpec` tests. It provides:
 - a fake projeciton implementation so I can run projections without a backing DB
 - an in-memory offset store
 - an in-memory telemetry and instruments (TODO: the instruments are currently global variables, instead, they should track data per projectionId so tsts can run in parallel again)